environment/src/main/java/com/sequenceiq/environment/exception/mapper/SearchCauseExceptionMapper.java

```diff
@@ -12,13 +14,29 @@
     @Override
     Response.Status getResponseStatus(T exception) {
         Response.Status defaultResponse = Response.Status.BAD_REQUEST;
-        if (exception.getCause() != null && Exception.class == exception.getCause().getClass()) {
+        if (exception == null || exception.getCause() != null && Exception.class == exception.getCause().getClass()) {
```

Comment:
It will return with default response, but doesn't this hide root causes where the root cause is encapsulated in an "Exception" class??